query_execution/QueryManager.cpp

```diff
@@ -0,0 +1,469 @@
+/**
+ *   Copyright 2016, Quickstep Research Group, Computer Sciences Department,
+ *     University of Wisconsin—Madison.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ **/
+
+#include "query_execution/QueryManager.hpp"
+
+#include <cstddef>
+#include <memory>
+#include <utility>
+#include <vector>
+
+#include "catalog/CatalogDatabase.hpp"
+#include "catalog/CatalogRelation.hpp"
+#include "catalog/CatalogTypedefs.hpp"
+#include "catalog/PartitionScheme.hpp"
+#include "query_execution/QueryExecutionMessages.pb.h"
+#include "query_execution/QueryExecutionTypedefs.hpp"
+#include "query_execution/WorkerMessage.hpp"
+#include "query_optimizer/QueryHandle.hpp"
+#include "relational_operators/RebuildWorkOrder.hpp"
+#include "relational_operators/WorkOrder.hpp"
+#include "storage/StorageBlock.hpp"
+#include "storage/StorageBlockInfo.hpp"
+
+#include "glog/logging.h"
+
+using std::pair;
+
+namespace quickstep {
+class CatalogDatabaseLite;
+class StorageManager;
+}
+
+namespace quickstep {
+
+QueryManager::QueryManager(const tmb::client_id foreman_client_id,
+                           const int num_numa_nodes,
+                           QueryHandle *query_handle,
+                           CatalogDatabaseLite *catalog_database,
+                           StorageManager *storage_manager,
+                           tmb::MessageBus *bus)
+      : foreman_client_id_(foreman_client_id),
+        query_id_(DCHECK_NOTNULL(query_handle)->query_id()),
+        catalog_database_(DCHECK_NOTNULL(catalog_database)),
+        storage_manager_(DCHECK_NOTNULL(storage_manager)),
+        bus_(DCHECK_NOTNULL(bus)) {
+  DCHECK_NOTNULL(query_handle->getQueryPlanMutable());
+  query_dag_ = query_handle->getQueryPlanMutable()->getQueryPlanDAGMutable();
+  DCHECK_NOTNULL(query_dag_);
+
+  const dag_node_index num_operators_in_dag = query_dag_->size();
+
+  output_consumers_.resize(num_operators_in_dag);
+  blocking_dependencies_.resize(num_operators_in_dag);
```

Comment:
This is optionally, but I prefer to construct them with explicit size, instead of doing a later `resize`.

So, I suggest the following refactor to move above two in the init list of the constructor:
```
  query_dag_(...),
  output_consumers_(query_dag_->size()),
  blocking_dependencies_(query_dag_->size()),
  ... {
```